answer/Explanation.java

```diff
@@ -32,14 +34,21 @@
 public class Explanation {
 
     private final List<ConceptMap> answers;
+    private final Rule rule;
 
-    public Explanation(List<ConceptMap> ans) {
+    public Explanation(List<ConceptMap> ans, Rule rule) {
```

Comment:
can put `@nullable` infront of `Rule rule` here, just to indicate the intent is that it _may_ be null